### ISSUE TITLE
Annotations: `$__interval` and `$__rate_interval` are incorrectly calculated

### ISSUE THREAD

**Issue by piotrminkina** [grafana/grafana#82636]
### What happened?

I am trying to use the `$__interval` and `$__rate_interval` variables in PromQL queries, in the annotation configuration. Looking up the results in the DevTools tool, I noticed in the response that these values are miscalculated.

When such a request is sent to the `/api/ds/query` URL:

```json
{
  "from": "1707980400000",
  "to": "1708009200000",
  "queries": [
    {
      "expr": "timestamp((kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} - kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} offset $__interval) > 0) * 1000",
      "range": true,
      "instant": false,
      "exemplar": false,
      "interval": "15s",
      "refId": "X",
      "datasource": {
        "type": "prometheus",
        "uid": "prometheus"
      },
      "legendFormat": ""
    }
  ]
}
```

In response, Grafana returns such data:

```json
{
  "results": {
    "X": {
      "status": 200,
      "frames": [
        {
          "schema": {
            "refId": "X",
            "meta": {
              "typeVersion": [
                0,
                0
              ],
              "executedQueryString": "Expr: timestamp((kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} - kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} offset 1s) > 0) * 1000\nStep: 5m0s"
            },
            "fields": []
          },
          "data": {
            "values": []
          }
        }
      ]
    }
  }
}
```

In the response, you can see that the `$__interval` variable was calculated for `1s`, which is clearly not true for a step of 5m0s. I also sent a similar request as above, but replacing `$__interval` with `$__rate_interval` and the response is as follows:

```json
{
  "results": {
    "X": {
      "status": 200,
      "frames": [
        {
          "schema": {
            "refId": "X",
            "meta": {
              "typeVersion": [
                0,
                0
              ],
              "executedQueryString": "Expr: timestamp((kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} - kube_pod_container_status_restarts_total{cluster=\"\", namespace=\"amo-dev\", pod=\"grafana-image-renderer-6fc7585cbd-g9ltl\", container=~\".+\"} offset 1m0s) > 0) * 1000\nStep: 5m0s"
            },
            "fields": []
          },
          "data": {
            "values": []
          }
        }
      ]
    }
  }
}
```

Here we see that the `$__rate_interval` variable is also miscalculated, since for a step of 5m0s it should be at least 20m0s.

### What did you expect to happen?

I expect these variables to be calculated correctly.

### Did this work before?

I don't know if this feature worked properly in previous versions of Grafana.

### How do we reproduce it?

1. Configure dashboard annotations using Prometheus as Data source. Use variables `$__interval` and `$__rate_interval` in queries.
2. Launch the developer tools in your browser.
3. Analyze the queries made to Grafana when you enable the configured annotations.

### Environment (with versions)?

Grafana: v10.3.3 (252761264e)
OS: Ubuntu 22.04.3 LTS
Browser: Google Chrome Version 121.0.6167.85 (Official Build) (64-bit)

### Grafana platform?

Kubernetes

### Datasource(s)?

Prometheus 2.29.1

**Comment by itsmylife** [comment]
The [PR](https://github.com/grafana/grafana/pull/77234) (And the PR description is explaining the reasons) fixed this issue. 
After having that PR we realized there were some small bugs and those were fixed with 
- https://github.com/grafana/grafana/pull/78193
- https://github.com/grafana/grafana/pull/80904

I am closing this issue since the fixes are merged. Please feel free to open a new one if you face another issue or reopen this one.

**Comment by itsmylife** [comment]
@dpericaxon Sorry I wasn't quite clear. 
Yes the fixes is in `v10.4.0`.

If a `minStep` exists then we use this for calculation. Otherwise, we use scrape interval. 
Let's take this case

Scrape Interval: `30s`
Interval the one Grafana calculated based on the time range and the panel width is `20s`
And we specify `minStep` as `1m`

Since we have the `minStep` we will use it. 

```
$__rate_interval Formula: max($__interval + Scrape interval, 4 * Scrape interval)
max(20s + 1m, 4 * 1m) => 4m
```

If we don't specify the `minStep`

```
$__rate_interval Formula: max($__interval + Scrape interval, 4 * Scrape interval)
max(20s + 30s, 4 * 30s) => 120s
```
